tracer/src/Datadog.Trace/PDBs/DatadogPdbReader.cs

```diff
@@ -74,9 +74,9 @@ private bool TryGetSymbolMethodOfAsyncMethod(MethodDef mdMethod, out SymbolMetho
             var stateMachineAttributeFullName = typeof(AsyncStateMachineAttribute).FullName;
             var stateMachineAttributeOfMdMethod = mdMethod.CustomAttributes?.FirstOrDefault(ca => ca.TypeFullName == stateMachineAttributeFullName);
             // Grab the generated inner type state machine from the argument given by the compiler to the AsyncStateMachineAttribute
-            var stateMachineInnerType = (stateMachineAttributeOfMdMethod?.ConstructorArguments.FirstOrDefault().Value as ClassSig)?.TypeDef;
+            var stateMachineInnerTypeTypeDefOrRefSig = (stateMachineAttributeOfMdMethod?.ConstructorArguments.FirstOrDefault().Value as TypeDefOrRefSig);
```

Comment:
LGTM - but a bit concerned that we don't have any test coverage for this code path.  Could you add a test for the scenario where you caught this?

Reply:
It happens in Optimized Code... it will be covered once we will add support for optimized / non-optimized test execution that @shurivich is currently working on.

Reply:
That's fair. Resolving.